pkg/composableschemadsl/compiler/translator.go

```diff
@@ -58,6 +58,14 @@ func translate(tctx translationContext, root *dslNode) (*CompiledSchema, error)
 	// as we do our walk
 	names := tctx.existingNames.Copy()
 
+	// Do an initial pass to translate partials and add them to the
+	// translation context. This ensures that they're available for
+	// subsequent reference in definition compilation.
+	err := translatePartials(tctx, root)
```

Comment:
call this `collectPartials`, since it isn't actually translating them into the root of the tree